src/nvim/tui/input.c

```diff
@@ -134,12 +128,44 @@ static void tinput_wait_enqueue(void **argv)
   uv_mutex_unlock(&input->key_buffer_mutex);
 }
 
+static void tinput_paste_event(void **argv)
+{
+  static bool active = false;
+  static bool canceled = false;
+
+  String keys = { .data = argv[0], .size = (size_t)argv[1] };
+  intptr_t phase = (intptr_t)argv[2];
+  if (active && phase == 1) {
```

Comment:
Seems like `active` could be eliminated by checking phase instead.

Reply:
How? `handle_bracketed_paste` does not set `phase = 2 `. You mean we should try to move this logic to TUI thread?

Reply:
If that simplifies it (eliminates the extra flags) that would be nice.